src/tree.rs

```diff
@@ -1141,6 +1145,41 @@ fn detect_cycles(parents: &[ResolvedParent]) -> Option<Index> {
     None
 }
 
+fn break_cycles(builder: &mut Builder, cycle_start_index: Index, parent_index: Index) -> Result<()> {
+    let mut unfiled = Item::new(UNFILED_GUID, Kind::Folder);
+    unfiled.age = 0;
+    unfiled.needs_merge = true;
+
+    let entries = &mut builder.entries;
+
+    let cycle_start_guid = &entries[cycle_start_index]
+        .item
+        .clone()
+        .guid;
+
+    let parent_guid = entries[parent_index]
+        .item
+        .clone()
+        .guid;
+
+    builder
+        .item(unfiled)?
+        .by_children(&ROOT_GUID)?;
+
+    builder
+        .parent_for(&parent_guid)
+        .by_children(&UNFILED_GUID)?;
+
+    builder
+        .parent_for(&cycle_start_guid)
+        .by_children(&parent_guid.clone())?;
+
```

Comment:
The builder calls above seem to create the relationships I want, but they don't remove the old cycles. Should they or will they be resolved after the builder is marked diverged?